mainframe/src/main/java/com/PiXl/mainframe/entities/UserEntity.java

```diff
@@ -0,0 +1,41 @@
+package com.PiXl.mainframe.entities;
+
+import com.PiXl.mainframe.Models.User;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import jakarta.persistence.EmbeddedId;
+import jakarta.persistence.Entity;
+import jakarta.persistence.Table;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Entity()
+@Table(name = "user")
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@ToString
+public class UserEntity {
+    @EmbeddedId
+    private UserEntityId id;
```

Comment:
Also, is there a specific reason for using an embedded ID for UserId?

Reply:
Alright. Other than this, everything looks fine to me.
I will resolve this conversation and approve the merge request from my end.

Reply:
I have resolved this issue. will try to keep in mind to revert my testing changes before pushing next time.